docs/content/how-to/disconnected/automatically-initialize-registry-overrides.md

```diff
@@ -0,0 +1,23 @@
+# Automatically Initialize RegistryOverrides with Image Content Type Policies
+## General
+The HyperShift Operator (HO) will automatically initialize the control plane operator (CPO) with any image registry override information from any ImageContentSourcePolicy (ICSP) or any ImageDigestMirrorSet (IDMS) instances from an OpenShift management cluster.
+
+!!! note
+
+    OpenShift management clusters do not allow both ICSP and IDMS CR instances.
+    IDMS CRs should be used with OpenShift release image versions 4.13 or higher.
+    ICSPs will be deprecated in future OpenShift release versions.
+
+## Technical Implementation Details
+### General
+In `hypershift-operator/main.go`, the HO will look to see if its management cluster has either ICSP or IDMS capabilities. If so, it will retrieve the image registry information from the appropriate ICSP or IDMS instances and store them in a variable called `imageRegistryOverrides`. This variable is then provided to a custom release image provider called `ProviderWithOpenShiftImageRegistryOverrides`.
+
+### ProviderWithOpenShiftImageRegistryOverrides
+The `Lookup` function for `ProviderWithOpenShiftImageRegistryOverrides` will use the source and mirror information in `imageRegistryOverrides` when attempting to look up release images.
+The `ProviderWithOpenShiftImageRegistryOverrides` is also provided to the `HostedClusterReconciler` as its `ReleaseProvider`. When the `HostedClusterReconciler` is reconciling, it will pass any image registry overrides to the ignition server reconciler and the CPO deployment specification.
+
+### Ignition Server
+The ignition server reconciler forwards this information on to the `ignition-server` container as an environment variable called `OPENSHIFT_IMG_OVERRIDES`. `hypershift/ignition-server/cmd/start.go` retrieves this information for the `ReleaseProvider` for the `TokenSecretReconciler`. An important caveat for the ignition server, it cannot follow the ImageContentSourcePolicy or ImageDigestMirrorSet rules because there is not a runtime running inside the pod to do the transformations. So, it's necessary to do the image URL live translation to the custom registry address.
```

Comment:
> An important caveat for the ignition server, it cannot follow the ImageContentSourcePolicy or ImageDigestMirrorSet rules because there is not a runtime running inside the pod to do the transformations. So, it's necessary to do the image URL live translation to the custom registry address.

Not sure I follow this explanation. The ignition server accesses the MCO, CCO images directly from their source registry which is why it needs to be aware of any mirror mappings.

Reply:
That's right, what I meant is, even if you set the `/etc/containers/registries.d/icsp-XXXX.conf`, the ignittion server cannot make the translation by himself, this is why we need to map both operators and map them using the `--registries-override=originalSourceImage=privateRegistryImage`